lib/Inspection/StatusT.h

```diff
@@ -0,0 +1,95 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <type_traits>
+
+#include <optional>
+
+#include "Inspection/Status.h"
+
+namespace arangodb::inspection {
+
+template<typename T>
+struct StatusT {
+  using Contained = std::variant<Status, T>;
+
+  auto static error(Status&& status) -> StatusT<T> {
+    return StatusT(
+        std::move(Contained(std::in_place_index<0>, std::move(status))));
```

Comment:
Why the extra move? This should suffice:
```suggestion
    return StatusT(
        Contained(std::in_place_index<0>, std::move(status)));
```